app/Jobs/PollForMediaWikiJobsJob.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Wiki;
+use Illuminate\Support\Facades\Http;
+
+class PollForMediaWikiJobsJob extends Job
+{
+    public function handle(): void
+    {
+        $wikis = Wiki::all()->pluck('domain');
+        foreach ( $wikis as $wikiDomain ) {
+            if ($this->hasPendingJobs( $wikiDomain )) {
+                $this->enqueueWiki( $wikiDomain );
+            }
+        }
+    }
+
+    private function hasPendingJobs( string $wikiDomain ): bool
+    {
+        $response = Http::withHeaders([
+            'host' => $wikiDomain
+        ])->get(
+            getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php?action=query&meta=siteinfo&siprop=statistics&format=json'
+        );
+
+        if ($response->failed()) {
+            $this->fail(
+                new \RuntimeException(
+                    'Failure polling wiki '.$wikiDomain.' for pending MediaWiki jobs: '.$response->clientError()
+                )
+            );
+
+            return false;
+        }
+
+        return $response->json()['query']['statistics']['jobs'] > 0;
```

Comment:
In PHP, is there an idiomatic way of safely accessing a nested array without the possibility of a Runtime Exception thrown on malformed data? I.e. how could I make sure this gracefully handles a json response of `{'error': 'Bad Gateway'}`

Reply:
I believe php doesn't really have this without quite some ugly looking stuff. Laravel provides https://laravel.com/docs/5.5/helpers#method-data-get which I would probably use here